lib/certificate.js

```diff
@@ -0,0 +1,142 @@
+import crypto from 'crypto';
+import B from 'bluebird';
+import path from 'path';
+
+const sqlite3 = B.promisifyAll(require('sqlite3'));
+const openssl = B.promisify(require('openssl-wrapper').exec);
+
+let tset = `<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n
+    <!DOCTYPE plist PUBLIC \"-//Apple//DTD PLIST 1.0//EN\" \"http://www.apple.com/DTDs/PropertyList-1.0.dtd\">
+    <plist version=\"1.0\">
+    <array/>
+</plist>`;
+
+/**
+ * Library for programatically adding certificates 
+ */
+class Certificate {
+
+  constructor (pemFilename) { 
+    this.pemFilename = pemFilename;
+  }
+
+  /**
+   * Add a certificate to the TrustStore
+   */
+  async add (dir) {
+    let data = await this.getDerData(this.pemFilename);
+    let subject = await this.getSubject(this.pemFilename);
+    let fingerprint = await this.getFingerPrint(this.data);
+
+    let trustStore = new TrustStore(dir);
+    return trustStore.addRecord(fingerprint, tset, subject, data);
+  }
+
+  /**
+   * Checks if keychain at given directory has this certificate
+   */
+  async has (dir) {
+    let subject = await this.getSubject(this.pemFilename);
+
+    let trustStore = new TrustStore(dir);
+    let records = await trustStore.getRecords(subject);
+    return records.length > 0;
+  }
+
+  /**
+   * Remove certificate from the TrustStore
+   */
+  async remove (dir) { 
+    let subject = await this.getSubject(this.pemFilename);
+    let trustStore = new TrustStore(dir);
+    return trustStore.removeRecord(subject);
+  }
+
+  /**
+   * Translate PEM file to DER buffer
+   */
+  async getDerData () {
+    if (this.data) {
+      return this.data;
+    }
+
+    // Convert 'pem' file to 'der'
+    return openssl('x509', {
```

Comment:
I'm having it set ```this.data = return openssl(...)```

The 'getDerData', 'getSubj', and 'getFingerprint' methods cache the results and then return them if they've been calculated already. I wanted lazy evaluation so that it doesn't do any unnecessary OpenSSL operations